demisto_sdk/commands/validate/validate_manager.py

```diff
@@ -0,0 +1,841 @@
+    incidentfield = 6
+    indicatorfield = 7
+    reputation = 8
+    layout = 9
+    dashboard = 10
+    incidenttype = 11
+    mapper = 12
+    classifier_5_9_9 = 13
+    classifier = 14
+    widget = 15
+    report = 16
+    connection = 17
+    readme = 18
+    releasenotes = 19
+
+
+SKIPPED_FILE_TYPES = ('changelog', 'description', 'testplaybook')
+
+
+class ValidateManager:
+    def __init__(self, is_backward_check=True, prev_ver=None, use_git=False, only_committed_files=False,
+                 print_ignored_files=False, skip_conf_json=True, validate_id_set=False, file_path=None,
+                 validate_all=False, is_external_repo=False, skip_pack_rn_validation=False, print_ignored_errors=False,
+                 silence_init_prints=True):
+        self.handle_error = BaseValidator(print_as_warnings=print_ignored_errors).handle_error
+        self.print_ignored_errors = print_ignored_errors
+        self.skip_docker_checks = False
+        self.skip_conf_json = skip_conf_json
+        self.is_backward_check = is_backward_check
+        self.validate_id_set = validate_id_set
+        self.is_circle = only_committed_files
+        self.validate_all = validate_all
+        self.file_path = file_path
+        self.branch_name = ''
+        self.use_git = use_git
+        self.skip_pack_rn_validation = skip_pack_rn_validation
+        self.changes_in_schema = False
+        self.check_only_schema = False
+        self.prev_ver = prev_ver if prev_ver else 'origin/master'
+        self.print_ignored_files = print_ignored_files
+        self.added_rn = set()
+        self.ignored_files = set()
+
+        if is_external_repo:
+            click.echo('Running in a private repository')
+            self.skip_conf_json = True
+
+        if validate_all:
+            self.skip_docker_checks = True
+            self.skip_pack_rn_validation = True
+
+        if self.validate_id_set:
+            self.id_set_validator = IDSetValidator(is_circle=self.is_circle, configuration=Configuration())
+
+    def print_final_report(self, valid):
+        if valid:
+            self.print_ignored_files_report()
+            self.print_ignored_errors_report()
+            click.secho('\nThe files are valid', fg='green')
+            return 0
+        else:
+            all_failing_files = '\n'.join(FOUND_FILES_AND_ERRORS)
+            self.print_ignored_files_report()
+            self.print_ignored_errors_report()
+            click.secho(f"\n=========== Found errors in the following files ===========\n\n{all_failing_files}\n",
+                        fg="bright_red")
+            click.secho('The files were found as invalid, the exact error message can be located above',
+                        fg='red')
+            return 1
+
+    def run_validation(self):
+        """Initiates validation in accordance with mode (i,g,a)
+        """
+        if self.validate_all:
+            return self.print_final_report(self.run_validation_on_all_packs())
```

Comment:
```suggestion
            result = self.run_validation_on_all_packs()
```

and then in the end, use the self.print_final_report(..)